src/backend/distributed/planner/multi_router_planner.c

```diff
@@ -488,6 +490,52 @@ ExtractInsertRangeTableEntry(Query *query)
 }
 
 
+static bool
+HasTIDColumn(Node *node)
+{
+	List *queryList = NIL;
+	ListCell *queryCell = NULL;
+
+	ExtractQueryWalker(node, &queryList);
```

Comment:
Since we only call this function with a Query *, maybe you could change it to `QueryHasTidColumn(Query *query)` and avoid the `ExtractQueryWalker` call.

Reply:
I want to reach jointree at every level. How can I do that without `ExtractQueryWalker`?

Reply:
Why do you want to do that? I thought the CTID that we wanted to ignore is only in the target list at the top level?